src/Releasy/Development/Model.fs

```diff
@@ -0,0 +1,25 @@
+module Releasy.Development.Model
+
+open System
+
+type MergeRequest = 
+    {
+        title: string
+        description: string
+        url: Uri
+    }
+
+type MRLinkedToFeatureEvent =
+    {
+      mergeRequest: MergeRequest
+      featureIdentifier: string
+    }
+
+type MRNotLinkedToAnyFeatureEvent =
+    {
+      mergeRequest: MergeRequest
+    }
+
+type MRLinkedEvent =
+    | MRLinkedToFeature of MRLinkedToFeatureEvent
+    | MRNotLinkedToAnyFeature of MRNotLinkedToAnyFeatureEvent
```

Comment:
So, a couple of small remarks on the types.
I would drop the `Event` suffix of the types because your discriminated union already shows that you are talking about an Event, so the types are just there to support that.

Also, creating a type with only one record is a bit of extra work that you don't really have to do.

I would write it like this:
```
type MRLinkedEvent =
    | MRLinkedToFeature of MRFeatureLink
    | MRNotLinkedToFeature of MergeRequest
```

This is a bit more expressive in language: either you have MRLinkedEvent of type MRLinkedToFeature that takes an MRFeatureLink or you have an MRNotLinkedToFeature that takes a MergeRequest.

If you want to be even more expressive, you could go for this:
```
type MRLinkedEvent =
    | MRLinkedToFeature of MRFeatureLink
    | MRNotLinkedToFeature of MergeRequest

and MRFeatureLink =
    { ... }
```
that way it is very clear that the MRFeatureLink is part of the DU.

This is a very good resource on the topic:https://github.com/thinkbeforecoding/FsUno
I hope this helps :)

Reply:
oh ! thanks for this feedback ! 
And we will end on Jeremy's blog :D